lib/Jsrt/JsrtInternal.h

```diff
@@ -152,32 +193,95 @@ JsErrorCode ContextAPIWrapper(Fn fn)
     }
     catch (Js::OutOfMemoryException)
     {
-      return JsErrorOutOfMemory;
+        errCode = JsErrorOutOfMemory;
     }
     catch (const Js::JavascriptException& err)
     {
         scriptContext->GetThreadContext()->SetRecordedException(err.GetAndClear());
-        return JsErrorScriptException;
+        errCode = JsErrorScriptException;
     }
     catch (Js::ScriptAbortException)
     {
         Assert(scriptContext->GetThreadContext()->GetRecordedException() == nullptr);
         scriptContext->GetThreadContext()->SetRecordedException(scriptContext->GetThreadContext()->GetPendingTerminatedErrorObject());
-        return JsErrorScriptTerminated;
+        errCode = JsErrorScriptTerminated;
     }
     catch (Js::EvalDisabledException)
     {
-        return JsErrorScriptEvalDisabled;
+        errCode = JsErrorScriptEvalDisabled;
+    }
+
+    CATCH_OTHER_EXCEPTIONS(errCode)
+
+    return errCode;
+}
+
+template <bool verifyRuntimeState, class Fn>
+JsErrorCode ContextAPIWrapper_WithRecord(Fn fn)
```

Comment:
Can we try to minimize the duplicate code here with a template parameter type for TTDRecorder, which can be a dummy no-op TTDREcorder if we don't want to record?